developerTools/scriptReferences/entityClientScript/entityClientScript.js

```diff
@@ -0,0 +1,116 @@
+//
+// entityClientScript.js
+//
+// Adapted from script https://github.com/robin-k-wilson/hifi-dev-utils/blob/master/client_entity_all_events.js
+// Updated by Robin Wilson 4/15/2019
+//
+// Generic client entity methods and signals that are triggered by mouse or controller interactions
+// Use as a template for your client entity scripts
+//
+// Client entity scripts are run on a interface instance. Meaning each avatar connected
+// to a domain have one instance of this script running locally and one client controls
+// an entity at one time called "ownership over that entity".
+// Best for physics changes.
+//
+// Copyright 2019 High Fidelity, Inc.
+//
+// Distributed under the Apache License, Version 2.0.
+// See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+
+(function() {
+    var I_AM_STATIC_VARIABLE = 200;
+    var _this = null;
+    var _entityID = null;
+
+    function EntityClientScript() {
+        _this = this;
+    }
+
+    EntityClientScript.prototype = {
+        // REQUIRED FUNCTIONS
+        preload: function(entityID) {
+            // runs when this script is refreshed or a
+            // client connects to a domain where this entity is present
+            _entityID = entityID;
+        },
+
+        unload: function() {
+            // triggered when avatar leaves the domain where entity is present
+            // clear any intervals
+            // clear any listeners
+            // reset anything else that needs to be
+        },
+
+        // EVENT LISTENERS FOR THE ENTITY
+
+        // Entity Event Listener Notes: 
+        //     For mouse/trigger click on an entity to do an action
+        //         Use: startNearTrigger + startFarTrigger + clickReleaseOnEntity
+
+        // MOUSECLICK METHODS - not triggered by controller
```

Comment:
i thought some of these _were_ triggered by hand controllers - am i wrong about that (i easily could be)?

Reply:
You're right some of them work. I'll have to do some investigation on which ones.